nexus/db-queries/src/db/datastore/region_snapshot_replacement.rs

```diff
@@ -93,27 +92,32 @@ impl DataStore {
         request: RegionSnapshotReplacement,
         volume_id: Uuid,
     ) -> Result<(), Error> {
+        let err = OptionalError::new();
         let conn = self.pool_connection_authorized(opctx).await?;
 
         self.transaction_retry_wrapper(
             "insert_region_snapshot_replacement_request_with_volume_id",
         )
         .transaction(&conn, |conn| {
             let request = request.clone();
+            let err = err.clone();
             async move {
                 use db::schema::region_snapshot_replacement::dsl;
-                use db::schema::volume_repair::dsl as volume_repair_dsl;
 
-                // An associated volume repair record isn't _strictly_ needed:
-                // snapshot volumes should never be directly constructed, and
-                // therefore won't ever have an associated Upstairs that
-                // receives a volume replacement request. However it's being
-                // done in an attempt to be overly cautious.
-
-                diesel::insert_into(volume_repair_dsl::volume_repair)
-                    .values(VolumeRepair { volume_id, repair_id: request.id })
-                    .execute_async(&conn)
-                    .await?;
+                // An associated volume repair record isn't _strictly_
+                // needed: snapshot volumes should never be directly
```

Comment:
It's not strictly speaking necessary - many replacements _could_ occur on the snapshot volume at the same time, and because it's never constructed there wouldn't be any repair operation required.

Reply:
They wouldn't contend on the snapshot volume's database record?

Reply:
They would only contend around the volume repair record. If there was no lock for the snapshot volume, then the individual replacement transactions could all fire in whatever order they're going to serialize in, and it would probably work.